addons/mail/static/src/components/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -1464,6 +1464,231 @@ QUnit.test('chat window: switch on TAB', async function (assert) {
+        document.activeElement,
+        "after pressing tab on the chat window on its left, the last chatWindow should have focus"
+    );
+
+    await afterNextRender(() =>
+        triggerEvent(
+            chatWindows[0].querySelector('.o_ComposerTextInput_textarea'),
+            'keydown',
+            { key: 'Tab' },
+        )
+    );
+    assert.strictEqual(
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "the leftmost chatWindow should have the focus after pressing tab on the rightmost chat window"
+    );
+});
+
+QUnit.test('chat window: cycle between more than 2 chat windows (RTL)', async function (assert) {
+    /**
+     * computation uses following info:
+     * ([mocked] global window width: @see `mail/static/src/utils/test_utils.js:start()` method)
+     * (others: @see mail/static/src/models/chat_window_manager/chat_window_manager.js:visual)
+     *
+     * - chat window width: 325px
+     * - start/end/between gap width: 10px/10px/5px
+     * - hidden menu width: 200px
+     * - global width: 1920px
+     *
+     * Enough space for 2 visible chat windows:
+     *  10 + 325 + 5 + 325 + 10 = 670 < 1920
+     */
+    assert.expect(6);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: "channel",
+                id: 1,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel1",
+                state: 'open',
+            }, {
+                channel_type: "channel",
+                id: 2,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel2",
+                state: 'open',
+            }, {
+                channel_type: "channel",
+                id: 3,
+                is_minimized: true,
+                is_pinned: true,
+                name: "channel3",
+                state: 'open',
+            }],
+        },
+    });
+    await this.start({
+        env: {
+            browser: {
+                innerWidth: 1920,
+            },
+        },
+    });
+    this.env.messaging.locale.update({
+        textDirection: 'rtl'
+    });
+    assert.containsN(
+        document.body,
+        '.o_ChatWindow .o_ComposerTextInput_textarea',
+        3,
+        "initialy, 3 chat windows should be present"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_ChatWindow.o-folded',
+        "all 3 chat windows should be open"
+    );
+
+    const chatWindows = document.querySelectorAll('.o_ChatWindow');
+    await afterNextRender(() => {
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea').focus();
+    });
+    assert.strictEqual(
+        chatWindows[2].querySelector('.o_ComposerTextInput_textarea'),
+        document.activeElement,
+        "The first chatWindow should have the focus"
```

Comment:
first -> rightmost (?) 
Which one is actually selected here?

For me the rest of this behavior is not correct as already pointed in other comments.